src/main/java/uk/gov/hmcts/juror/api/moj/service/report/YieldPerformanceReportServiceImpl.java

```diff
@@ -0,0 +1,186 @@
+package uk.gov.hmcts.juror.api.moj.service.report;
+
+import com.querydsl.core.Tuple;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.juror.api.moj.controller.reports.request.YieldPerformanceReportRequest;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.AbstractReportResponse;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.YieldPerformanceReportResponse;
+import uk.gov.hmcts.juror.api.moj.exception.MojException;
+import uk.gov.hmcts.juror.api.moj.repository.CourtQueriesRepository;
+import uk.gov.hmcts.juror.api.moj.repository.IPoolCommentRepository;
+import uk.gov.hmcts.juror.api.moj.repository.JurorPoolRepository;
+import uk.gov.hmcts.juror.api.moj.utils.SecurityUtil;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.Math.round;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class YieldPerformanceReportServiceImpl implements YieldPerformanceReportService {
+    private final JurorPoolRepository jurorPoolRepository;
+    private final IPoolCommentRepository poolCommentRepository;
+    private final CourtQueriesRepository courtQueriesRepository;
+
+
+    @Override
+    public YieldPerformanceReportResponse viewYieldPerformanceReport(
+        YieldPerformanceReportRequest yieldPerformanceReportRequest) {
+
+        YieldPerformanceReportResponse response = setupResponseHeaders(yieldPerformanceReportRequest);
+
+        List<String> courtLocCodesList = yieldPerformanceReportRequest.isAllCourts()
+            ? courtQueriesRepository.getAllCourtLocCodes(false)
+            : yieldPerformanceReportRequest.getCourtLocCodes();
+
+        String courtLocCodes = String.join(",", courtLocCodesList);
+
+        try {
+            List<String> yieldPerformanceReportStats =
+                jurorPoolRepository.getYieldPerformanceReportStats(courtLocCodes,
+                    yieldPerformanceReportRequest.getFromDate(),
+                    yieldPerformanceReportRequest.getToDate());
```

Comment:
Would be better to return an object with named variables

Reply:
This is how the postgres function returns tuples, maybe look into this as an ehancement

Reply:
Can we map the tuple into an object at point of consumption then use this object for our processing?

Reply:
updated to use objects from repository layer